### PR TITLE
Add Nix flake with package and NixOS module

### LINKED ISSUE #22: Add Nix Flake
Should provide:
- Package
- Development environment
- NixOS Module with watchdog config support
	- Bonus points if you can pass a derivation to be used as the server package

Might require some changes to the watchdog, such as:
- ~~Overriding instances directory location.~~ Option for this already exists!
- Optional `ApiTokenFile` config which reads the API token from a file for easier secret support when using sops-nix, etc

### PR DESCRIPTION
**Pull request by Zumorica**
Closes #22 
- Adds an option to read an instance's API Token from a file.
- Add a Nix flake
   - Includes a package for building the watchdog.
   - Includes a NixOS module for the watchdog.
       - appsettings.yml is configured through nix
       - Instances' config.toml can optionally be configured through nix as well
   - Includes a test VM definition!

### CODE REVIEW

**Comment by Zumorica** [comment]
@PJB3005 Could you look at the C# changes of this PR sometime? Just want to make sure what I added is a good idea.

File: SS14.Watchdog/Components/ServerManagement/ServerInstance.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ public sealed class ServerInstance : IServerInstance
 
         public string Key { get; }
         public string? Secret { get; private set; }
-        public string? ApiToken => _instanceConfig.ApiToken;
+        public string? ApiToken => _instanceConfig.ApiTokenFile == null
```
Comment (PJB3005) [inline SS14.Watchdog/Components/ServerManagement/ServerInstance.cs:34]: Don't like the fact this has to re-open the file every time the property is read.

Why can't you pass this in the regular config file?

Comment (Zumorica) [reply SS14.Watchdog/Components/ServerManagement/ServerInstance.cs:34]: > Don't like the fact this has to re-open the file every time the property is read.

Should I just cache it on the constructor?

> Why can't you pass this in the regular config file?

This is much more convenient when handling secrets with sops-nix as it allows me to keep the watchdog settings on my server public. Here's an example from my NixOS config where my wireguard private key is read from a file: https://github.com/Zumorica/GradientOS/blob/main/hosts/briah/wireguard.nix#L40

Comment (PJB3005) [reply SS14.Watchdog/Components/ServerManagement/ServerInstance.cs:34]: > This is much more convenient when handling secrets with sops-nix as it allows me to keep the watchdog settings on my server public.

Can't you read the secret at evaluation time of the Nix code?

> Should I just cache it on the constructor?

I guess, if it's unavoidable.

Comment (Zumorica) [reply SS14.Watchdog/Components/ServerManagement/ServerInstance.cs:34]: > Can't you read the secret at evaluation time of the Nix code?

Not really, sadly. It's a bit of a limitation, but it's the safest way to do it with how Nix works.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**